x-pack/plugins/apm/server/plugin.ts

```diff
@@ -106,79 +106,43 @@ export class APMPlugin
 
     registerFeaturesUsage({ licensingPlugin: plugins.licensing });
 
-    const { ruleDataService } = plugins.ruleRegistry;
     const getCoreStart = () =>
       core.getStartServices().then(([coreStart]) => coreStart);
 
-    const alertsIndexPattern = ruleDataService.getFullAssetName(
-      'observability-apm*'
-    );
-
-    const initializeRuleDataTemplates = once(async () => {
-      const componentTemplateName = ruleDataService.getFullAssetName(
-        'apm-mappings'
-      );
-
-      if (!ruleDataService.isWriteEnabled()) {
-        return;
-      }
-
-      await ruleDataService.createOrUpdateComponentTemplate({
-        name: componentTemplateName,
-        body: {
-          template: {
-            settings: {
-              number_of_shards: 1,
-            },
-            mappings: mappingFromFieldMap(
-              {
-                [SERVICE_NAME]: {
-                  type: 'keyword',
-                },
-                [SERVICE_ENVIRONMENT]: {
-                  type: 'keyword',
-                },
-                [TRANSACTION_TYPE]: {
-                  type: 'keyword',
-                },
-                [PROCESSOR_EVENT]: {
-                  type: 'keyword',
-                },
+    const { ruleDataService } = plugins.ruleRegistry;
+    const ruleDataClient = ruleDataService.initializeIndex({
```

Comment:
does this also create the index?

Reply:
Looks like it doesn't, in that case maybe another name is in order? Something like `installResources`?

Reply:
@dgieselaar yes, it starts index bootstrapping right away inside the call in the background

Reply:
`installIndexResources` or `bootstrapIndex` sounds good to me, more descriptive names. I can rename it in a follow-up PR 👍

Reply:
first one sgtm 👍